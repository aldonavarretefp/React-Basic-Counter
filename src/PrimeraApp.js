// import React ,{Fragment} from 'react'
import React from 'react'
import PropTypes from 'prop-types';

const PrimeraApp = ( {saludo,otra = "No hay 'otra'"}) => {

    const saludoVar = "Hola mundo const"
    const isActive = false;
    const numeros = [1,2,3,4,5,6];
    const objeto = {
        nombre:"Aldo",
        edad:"20"
    }
    // if(!saludo){
    //     throw new Error("Saludo es necesario");
    // }
    return (
        <>
            <h1>
                Hola Mundo
            </h1>
            <p>{saludoVar}</p>
            <p>Hola mundo HTML</p>
            {
                isActive && 
                <h2>Active</h2>
            }
            <h1>{numeros}</h1>
            {/* {numeros.map(
                (el) =>(
                    <i> {el} </i>
                )
            )} */}
            <pre>{JSON.stringify(objeto,null,3)}</pre>
            <h1>{saludo}</h1>
            <h1>{otra }</h1>
            
        </>
    );
}

PrimeraApp.propTypes = {
    // saludo: PropTypes.string
    saludo: PropTypes.string.isRequired


}

PrimeraApp.defaultProps = {
    subtitulo:''
}


export default PrimeraApp
