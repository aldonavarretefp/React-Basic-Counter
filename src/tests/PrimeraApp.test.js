import React from 'react'
import {shallow} from 'enzyme';
import PrimeraApp from '../PrimeraApp';



describe('Pruebas en primera App',()=>{
    // test('debe mostrar Hola desde index', () => {
    //     const saludo  = 'Hola desde index';
    //     const {getByText} = render(<PrimeraApp saludo={saludo}/>);
    //     // eslint-disable-next-line testing-library/prefer-screen-queries
    //     expect(getByText(saludo)).toBeInTheDocument();
    // })
    
    test('debe mostrar <PrimeraApp /> correctamente', () => {
        const saludo = 'Hola desde index!';
        const wrapper = shallow(<PrimeraApp saludo ={saludo}/>);
        expect(wrapper).toMatchSnapshot(); 
    })

    test('debe mostrar el subtitulo enviado por props', () => {
        const saludo = 'Hola desde index!';
        const subtitulo = 'Subtitulo desde index';
        const wrapper = shallow(
            <PrimeraApp 
                saludo ={saludo}
                subtitulo = {subtitulo}
            />
        );

        const textoParrafo = wrapper.find('p');
        
        expect(textoParrafo.text()).toMatchSnapshot(subtitulo); 
    })
    
    
})