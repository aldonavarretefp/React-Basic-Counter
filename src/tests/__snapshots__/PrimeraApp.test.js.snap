// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en primera App debe mostrar <PrimerApp /> correctamente 1`] = `
<Fragment>
  <h1>
    Hola desde index!
    !
  </h1>
  <p />
</Fragment>
`;

exports[`Pruebas en primera App debe mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
    Hola desde index!
    !
  </h1>
  <p />
</Fragment>
`;

exports[`Pruebas en primera App debe mostrar el subtitulo enviado por props: Subtitulo desde index 1`] = `"Subtitulo desde index"`;
